src/random_access_file_type.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ /** @file random_access_file_type.h Class related to random access to files. */
+
+#ifndef RANDOM_ACCESS_FILE_TYPE_H
+#define RANDOM_ACCESS_FILE_TYPE_H
+
+#include "fileio_type.h"
+#include <string>
+
+/**
+ * A file from which bytes, words and double words are read in (potentially) a random order.
+ *
+ * This is mostly intended to be used for things that can be read from GRFs when needed, so
+ * the graphics but also the sounds. This also ties into the spritecache as it uses these
+ * files to load the sprites from when needed.
+ */
+class RandomAccessFile {
+	/** The number of bytes to allocate for the buffer. */
+	static const int BUFFER_SIZE = 512;
```

Comment:
This will probably cause issues (see #8962) - make it constexpr

Reply:
As far as I can see that issue was because it was used outside of the struct (i.e. it was public instead of it being private here), but I'll do it nevertheless as it doesn't seem to hurt.